### PR TITLE
Enable spin-separated 3-RDMs

### PR DESCRIPTION
**Pull request by brianz98**
The spin-separated 3-RDMs (aaa, aab, abb, bbb components) have been implemented thanks to @MatthewRHermes's hack (see #2326, code here is largely lifted out of https://github.com/MatthewRHermes/mrh/blob/master/my_pyscf/lassi/op_o0.py#L54-L168). A suitable test (mcscf/04) have been expanded to assert that the spin-separated RDMs add (with correct transpositions) to the spin-summed RDMs.

### CODE REVIEW

**Comment by MatthewRHermes** [comment]
You should probably add a test case to https://github.com/pyscf/pyscf/blob/master/pyscf/fci/test/test_rdm.py

**Comment by brianz98** [comment]
> You should probably add a test case to https://github.com/pyscf/pyscf/blob/master/pyscf/fci/test/test_rdm.py

Done. Thanks for the suggestion!

**Review by sunqm — APPROVED** [review]

**Comment by sunqm** [comment]
This PR looks good to me. @MatthewRHermes , any more comments?

File: examples/fci/14-density_matrix.py [hunk 1]
```diff
@@ -125,6 +125,18 @@
...
+#
+# Spin-separated 3-particle density matrix
+#
+
+(dm1a, dm1b), (dm2aa, dm2ab, dm2bb), (dm3aaa, dm3aab, dm3abb, dm3bbb) = \
+        fci.direct_spin1.make_rdm123s(fcivec0, norb, (nelec_a,nelec_b))
+
+assert(numpy.allclose(dm1a+dm1b, dm1))
+assert(numpy.allclose(dm2aa+dm2bb+dm2ab+dm2ab.transpose(2,3,0,1), dm2))
+assert(numpy.allclose(dm3aaa+dm3bbb+dm3aab+dm3aab.transpose(0,1,4,5,2,3)+\
```
Comment (MatthewRHermes) [inline examples/fci/14-density_matrix.py:137]: Maybe it's pedantic, but can we put this assert in test_rdm.py as well?

Comment (brianz98) [reply examples/fci/14-density_matrix.py:137]: done!

**Comment by MatthewRHermes** [comment]
LGTM

**Pull request merged**